tests/foreman/cli/test_hammer.py

```diff
@@ -41,31 +42,44 @@ def _traverse_command_tree(self, command):
         )
-        info = self._fetch_command_info(command)
-        for option in output['options']:
-            if (command == 'hammer sync-plan create' and
-                    option['name'] == 'sync-date'):
-                # sync-date option use the current server time as the default
-                # value in help text. As we use a previous created json file
-                # with the commands information we need a special treatment for
-                # this option.
-                info_option = None
-                for opt in info['options']:
-                    if opt['name'] == 'sync-date':
-                        info_option = opt
-                        break
-                self.assertIsNotNone(info_option)
-                self.assertEqual(option['name'], opt['name'])
-                self.assertEqual(option['shortname'], opt['shortname'])
-                self.assertEqual(option['value'], opt['value'])
-                # Drop the default value, and assert if the remaining string is
-                # present on the help text.
-                help = option['help'][:option['help'].find('Default: "')]
-                self.assertIn(help, opt['help'])
-            else:
-                self.assertIn(option, info['options'])
-        if len(info['subcommands']) > 0:
-            for subcommand in info['subcommands']:
+        command_options = set([option['name'] for option in output['options']])
+        command_subcommands = set(
+            [subcommand['name'] for subcommand in output['subcommands']]
+        )
+        expected = self._fetch_command_info(command)
+        expected_options = set()
+        expected_subcommands = set()
+
+        if expected is not None:
+            expected_options = set(
+                [option['name'] for option in expected['options']]
+            )
+            expected_subcommands = set(
+                [subcommand['name'] for subcommand in expected['subcommands']]
+            )
+
+        added_options = tuple(command_options - expected_options)
+        removed_options = tuple(expected_options - command_options)
+        added_subcommands = tuple(command_subcommands - expected_subcommands)
+        removed_subcommands = tuple(expected_subcommands - command_subcommands)
+
+        if (added_options or added_subcommands or removed_options or
+                removed_subcommands):
+            diff = {
+                'added_command': expected is None,
+            }
+            if added_options:
+                diff['added_options'] = added_options
+            if removed_options:
+                diff['removed_options'] = removed_options
+            if added_subcommands:
+                diff['added_subcommands'] = added_subcommands
+            if removed_subcommands:
+                diff['removed_subcommands'] = removed_subcommands
+            self.differences[command] = diff
+
+        if len(output['subcommands']) > 0:
```

Comment:
The the hammer subcommands instead of expected subcommands, this allows grabbing information for the added commands and subcommands.